mathics/builtin/numbers/calculus.py

```diff
@@ -1407,10 +1535,30 @@ class FindRoot(Builtin):
     }
 
     methods = {
+        "Automatic": find_root_newton,
         "Newton": find_root_newton,
         "Secant": find_root_secant,
     }
 
+    @staticmethod
+    def determine_epsilon(x0, options, evaluation):
+        """Determine epsilon  from a reference value, and from the accuracy and the precision goals"""
+        acc_goal, prec_goal = get_accuracy_and_prec(options, evaluation)
```

Comment:
My own personal opinion this is the kind of unnecessary coding is that I see a lot of which I find penny wise and pound foolish.

There are two independent parameters, acc_goal and prec_goal and each contributes to the returned value. We have this two if/else's embedded into in another if/else each with some sort of repeated expression given in Mathics expression - like format (i.e. is a little harder to understand than say a Python expression). 

A straightforward and easy-to-understand way to code this is simply not nest and duplicate expressions 

For example: 

```
# from_python(1e-10) is computed once outside and given a name and comment that describes its meaning 

if not (pres_goal or acc_goal): 
     return WHATEVER_FROM_1E_MINUS10_MEANS

eps = abs(x0) *  Integer10 ** (-prec_goal) if pres_goal else Integer0
if acc_goal:
   eps += Integer10 ** (-acc_goal)
return = apply_N(eps)
```

Reply:
@rocky,  OK, I like it! And no, it was not about performance, it was just about having a simplest routine that came to my mind that provides the output I needed. This one is much better!